policy/daemon.go

```diff
@@ -317,11 +316,28 @@ func (p *policyd) fetchPolicy(ctx context.Context, domain string) (*SignedPolicy
 		glg.Warn(errors.Wrap(err, "error body.close"))
 	}
 
-	// set eTag cache
-	eTag := res.Header.Get("ETag")
-	if eTag != "" {
-		glg.Debugf("Setting ETag %v for domain %s", eTag, domain)
-		p.etagCache.SetWithExpire(domain, &etagCache{eTag, sp}, p.etagExpTime)
+	// verify policy expiry
+	var etagValidDur time.Duration
+	if sp.SignedPolicyData.Expires == nil {
+		glg.Warnf("Warning: policy expiry not set, domain: %v", domain)
+		etagValidDur = p.etagExpTime - p.expireMargin
+	} else {
+		// if not set, sp.SignedPolicyData.Expires is nil
+		// if set, but invalid, sp.SignedPolicyData.Expires is Time{}
+		validDur := sp.SignedPolicyData.Expires.Time.Sub(fastime.Now())
+		if validDur <= p.expireMargin {
+			p.etagCache.Delete(domain)
```

Comment:
I think we do not need to delete the etag cache, just let it expire.

Reply:
availability V.S. correctness...

If the new policy or the cached policy is expired, it is logical to invalid all old caches...

Reply:
but this logic duplicates the logic with the etag cache?
https://github.com/yahoojapan/athenz-authorizer/pull/33/files#diff-05178a6859fde9a6cf69636df95e0aa5R333

Reply:
if the return response don't have etag, the old etag cache will not be updated.

Reply:
For performance wise, the old etag cache shouldn't be deleted and can be used in next fetch.

Reply:
block removed